tools/clang/lib/SPIRV/AstTypeProbe.cpp

```diff
@@ -1142,6 +1142,15 @@ bool isRelaxedPrecisionType(QualType type, const SpirvCodeGenOptions &opts) {
     }
   }
 
+  // Handles reference type, input/output variable for example.
+  {
+    QualType pointee = {};
+    if (type->isReferenceType() || type->isPointerType()) {
```

Comment:
For consistency with how we handle pointee types elsewhere in this file, can you change this to:

```
  // Reference types
  if (const auto *refType = type->getAs<ReferenceType>())
    return isRelaxedPrecisionType(refType->getPointeeType(), opts);

  // Pointer types
  if (const auto *ptrType = type->getAs<PointerType>())
    return isRelaxedPrecisionType(ptrType->getPointeeType(), opts);
```

Reply:
Hi @sudonatalie, I've committed your suggestions and added a test case to this pr.